File: admin/src/app/models/view-models/discount.ts
import { BaseViewModel } from "@app/models";
import { AutoMap } from "@automapper/classes";

export class Discount extends BaseViewModel {
    @AutoMap()
    id?: number;
    @AutoMap()
    percentDiscount: number
    @AutoMap()
    maxDiscountPrice: number
    @AutoMap()
    fromDate: Date
    @AutoMap()
    toDate: Date

}

File: App.Models/Entities/Discount.cs
using System;
using System.ComponentModel.DataAnnotations;
using MaxV.Common.Model;

namespace App.Models.Entities
{
    public class Discount : BaseEntity<int>
    {
        [Range(0, 100)]
        public int? PercentDiscount { get; set; }
        public double? MaxDiscountPrice { get; set; }
        public DateTime? FromDate { get; set; }
        public DateTime? ToDate { get; set; }
    }
}
